tools/webpack/blocks.js

```diff
@@ -0,0 +1,212 @@
+		'preformatted',
+		'pullquote',
+		'quote',
+		'separator',
+		'social-links',
+		'spacer',
+		'table',
+		'text-columns',
+		'verse',
+		'video',
+		...dynamicBlockFolders,
+	];
+	const blockPHPFiles = {
+		'widgets/src/blocks/legacy-widget/index.php': 'wp-includes/blocks/legacy-widget.php',
+		...dynamicBlockFolders.reduce( ( files, blockName ) => {
+			files[ `block-library/src/${ blockName }/index.php` ] = `wp-includes/blocks/${ blockName }.php`;
+			return files;
+		} , {} ),
+	};
+	const blockMetadataFiles = {
+		'widgets/src/blocks/legacy-widget/block.json': 'wp-includes/blocks/legacy-widget/block.json',
+		...blockFolders.reduce( ( files, blockName ) => {
+			files[ `block-library/src/${ blockName }/block.json` ] = `wp-includes/blocks/${ blockName }/block.json`;
+			return files;
+		} , {} ),
+	};
+
+	const blockPHPCopies = Object.keys( blockPHPFiles ).map( ( filename ) => ( {
+		from: join( baseDir, `node_modules/@wordpress/${ filename }` ),
+		to: join( baseDir, `src/${ blockPHPFiles[ filename ] }` ),
+	} ) );
+
+	const blockMetadataCopies = Object.keys( blockMetadataFiles ).map( ( filename ) => ( {
+		from: join( baseDir, `node_modules/@wordpress/${ filename }` ),
+		to: join( baseDir, `src/${ blockMetadataFiles[ filename ] }` ),
+	} ) );
+
+	const blockStylesheetCopies = blockFolders.map( ( blockName ) => ( {
+		from: join( baseDir, `node_modules/@wordpress/block-library/build-style/${ blockName }/*.css` ),
+		to: join( baseDir, `${ buildTarget }/blocks/${ blockName }/` ),
+		flatten: true,
+		transform: ( content ) => {
+			if ( mode === 'production' ) {
+				return postcss( [
+					require( 'cssnano' )( {
+						preset: 'default',
+					} ),
+				] )
+					.process( content, { from: 'src/app.css', to: 'dest/app.css' } )
+					.then( ( result ) => result.css );
+			}
+
+			return content;
+		},
+		transformPath: ( targetPath, sourcePath ) => {
+			if ( mode === 'production' ) {
+				return targetPath.replace( /\.css$/, '.min.css' );
+			}
+
+			return targetPath;
+		}
+	} ) );
+
+	const config = {
+		mode,
+		entry: {
+			'file/view': join( baseDir, `node_modules/@wordpress/block-library/build-module/file/frontend` ),
+		},
+		output: {
+			devtoolNamespace: 'wp',
+			filename: `[name]${ suffix }.js`,
+			path: join( baseDir, `${ buildTarget }/blocks` ),
+		},
+		resolve: {
+			modules: [
+				baseDir,
+				'node_modules',
+			],
+			alias: {
+				'lodash-es': 'lodash',
+			},
+		},
+		module: {
+			rules: [
+				{
+					test: /\.js$/,
+					use: [ 'source-map-loader' ],
+					enforce: 'pre',
+				},
+			],
+		},
+		optimization: {
+			moduleIds: mode === 'production' ? 'hashed' : 'named',
+		},
+		plugins: [
+			new DefinePlugin( {
+				// Inject the `GUTENBERG_PHASE` global, used for feature flagging.
+				'process.env.GUTENBERG_PHASE': 1,
+				'process.env.FORCE_REDUCED_MOTION': JSON.stringify(
+					process.env.FORCE_REDUCED_MOTION
+				),
+			} ),
+			new DependencyExtractionPlugin( {
+				injectPolyfill: true,
```

Comment:
It looks like the option for combining assets into one file doesn't quite work with multiple webpack configs so I left it disabled for blocks for now since we have only one file for now. It will require upstream changes.

The issue is that it only outputs only one file for the config that was processed last ...